src/services/codefixes/inferFromUsage.ts

```diff
@@ -335,22 +336,31 @@ namespace ts.codefix {
     }
 
     function inferTypeForVariableFromUsage(token: Identifier, program: Program, cancellationToken: CancellationToken): Type {
-        return InferFromReference.inferTypeFromReferences(getReferences(token, program, cancellationToken), program.getTypeChecker(), cancellationToken);
+        const references = getReferences(token, program, cancellationToken);
+        const checker = program.getTypeChecker();
+        const types = InferFromReference.inferTypesFromReferences(references, checker, cancellationToken);

```

Comment:
Seems like we could combine `inferTypesFromReferences` and `unifyFromContext` to avoid exposing two different functions?